src/batou/secrets/__init__.py

```diff
@@ -25,16 +25,23 @@
 
 from batou import DuplicateOverride, SuperfluousSecretsSection
 
-from .encryption import EncryptedConfigFile, iter_other_secrets
+from .encryption import (
+    EncryptedConfigFile,
+    get_secret_config_from_environment_name,
+    get_secrets_type,
+    iter_other_secrets,
+    secret_name_from_path,
+)
 
 
 def add_secrets_to_environment(environment):
-    secrets_file = (
-        pathlib.Path("environments") / environment.name / "secrets.cfg"
-    )
+    secrets_type = get_secrets_type(environment.name)
```

Comment:
Wording of "type", "provider", ... needs to be decided and unified. I don't particularly like "type" as that seems to be more indicative of which kind of secret we are encrypting, but not specific to the method we use for encryption.